src/main/java/com/android/volley/toolbox/BasicNetwork.java

```diff
@@ -178,7 +178,12 @@ public NetworkResponse performRequest(Request<?> request) throws VolleyError {
                 if (httpResponse != null) {
                     statusCode = httpResponse.getStatusCode();
                 } else {
-                    throw new NoConnectionError(e);
+                    if (request.shouldRetryConnectionErrors()) {
+                        attemptRetryOnException("no-connection", request, new NoConnectionError(e));
```

Comment:
nit: looks like Julien might have asked for this, but looking at how we label other connection errors, I think "connection" is probably more appropriate, as in the other cases we log the thing that is failing rather than the full failure (e.g. "server" instead of "server-error", "network" instead of "network-error", etc).

Reply:
sg, I just switched it back